src/library/scala/collection/immutable/Map.scala

```diff
@@ -192,6 +192,8 @@ object Map extends MapFactory[Map] {
     def updated[V1 >: V](key: K, value: V1): WithDefault[K, V1] =
       new WithDefault[K, V1](underlying.updated(key, value), defaultValue)
 
+    override def + [V1 >: V](kv: (K, V1)): WithDefault[K, V1] = updated(kv._1, kv._2)
```

Comment:
I think we should do the same with `++`. But I now remember that we’ve tried in #6943 and didn’t like it… So, for the sake of consistency, we should probably not do it with `+`, actually…

Reply:
Let me know if I can help once you decided how to fix https://github.com/scala/bug/issues/11314